### PR TITLE
Fix remote extension download v16

### PR DESCRIPTION
**Pull request by lubennikovaav**
When trying to download remote extension, strip pkglib_path prefix from request_path.
Normally, extensions should use `$libdir/` macro for this,
but for example pg_regress explicitly sets full path for its setup functions.

See details here:
https://github.com/neondatabase/neon/pull/8681/files#r1766508669

### CODE REVIEW

**Comment by hlinnaka** [comment]
Hmm, for pg_regress, the regress.so would be found locally, right? The remote extension download code should not be invoked in the first place. What am I missing?

**Comment by lubennikovaav** [comment]
> Hmm, for pg_regress, the regress.so would be found locally, right? The remote extension download code should not be invoked in the first place. What am I missing?

All the context in https://github.com/neondatabase/neon/pull/8681

TLDR: we put regress.so into custom extension to make it available for staging computes without including it into docker image for everyone

**Review by hlinnaka — APPROVED** [review]
> TLDR: we put regress.so into custom extension to make it available for staging computes without including it into docker image for everyone

ok

Do you need superuser to run the regression tests? If yes, you could probably do things like copy regress.so to the right directory with some SQL script that you run before the test.

**Comment by lubennikovaav** [comment]
> like copy regress.so to the right directory

Yes, but we need to put is somewhere first. Ideally, outside of compute image. And custom extensions fit this purpose
